arctic/store/version_store.py

```diff
@@ -576,6 +575,135 @@ def write(self, symbol, data, metadata=None, prune_previous_version=True, **kwar
         return VersionedItem(symbol=symbol, library=self._arctic_lib.get_name(), version=version['version'],
                              metadata=version.pop('metadata', None), data=None)
 
+    def _add_new_version_using_reference(self, symbol, new_version, reference_version, prune_previous_version):
+        if not new_version or \
+                not reference_version or \
+                new_version['symbol'] != reference_version['symbol'] or \
+                new_version['_id'] == reference_version['_id'] or \
+                not new_version['base_version_id']:
+            raise ArcticException("Illegal or non-matching nwe_version or reference_version were supplied: %s and %s",
```

Comment:
Belt and braces, but this could never happen?

Reply:
not in our case (and almost unlikely we get the same uuid from bson's rng).
Added so that it can be re-used safely in the future.

Reply:
I'd tend to user `assert` for constraints that can't be violated (apart from by bugs) and exceptions for things that are user API facing.

Reply:
changed to using assertion here